File: Code/TrustWords/levenshtein.py
words = []

def lev_distance(word1, word2):
    
    diff = 0

    len_diff = abs(len(word1) - len(word2))

    # Deals with different length strings
    if len(word1) > len(word2):
        min_len = len(word2)
        min_word = word1
        max_word = word2
    else:
        min_len = len(word1)
        min_word = word2
        max_word = word1

    for x in range(min_len):

        if min_word[x] != max_word[x]:
            diff += 1
    
    # Adds on the remaining letters
    diff += len_diff
    print(f"[*] Difference between: {min_word} and {max_word} is: {diff}")



with open("./en.csv") as file:
    data = file.readlines()

for line in data:
    parts = line.split(",")
    words.append(parts[2])

# Converts all strings to lower
words = list(map(str.lower, words))

for x in range(len(words) - 1):
    lev_distance(words[x], words[x + 1])

File: Code/SimilarWords/modes/levenshtein.cpp
#include <string>
#include <math.h>

int DIFFERENCE_TOLERANCE = 1;

/*
    Calculates the levistien distance of two words
*/
int lev_distance(std::string word1, std::string word2)
{
    int diff = 0;

    int len_diff = word1.length() - word2.length();
    len_diff = std::abs(len_diff);

    //Deals with different length strings
    int min_len = 0;
    std::string min_word, max_word;
    if (word1.length() > word2.length())
    {
        min_len = word2.length();
        min_word = word1;
        max_word = word2;
    }
    else
    {
        min_len = word1.length();
        min_word = word2;
        max_word = word1;
    }
        
    for (size_t i = 0; i < min_len; i++)
    {
        if (min_word[i] != max_word[i])
        {
            diff += 1;
        }
    }

    // Adds on the remaining letters
    diff += len_diff;
    return diff;
        
}

/*
    Returns true if the two entered words are deemed similar
*/
bool levenshtein_similar(std::string word1, std::string word2)
{
    int diff = lev_distance(word1, word2);
    return diff <= DIFFERENCE_TOLERANCE;
}